[BUG] test_spark_from_json_date_with_format FAILED on : Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec

**Describe the bug**

test_spark_from_json_date_with_format FAILED on : Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec,

Related PR: https://github.com/NVIDIA/spark-rapids/pull/10490


```
 FAILED ../../src/main/python/json_test.py::test_spark_from_json_date_with_format[DATAGEN_SEED=1709762464, INJECT_OOM] - pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec


=================================== FAILURES ===================================
____________________ test_spark_from_json_date_with_format _____________________
[gw3] linux -- Python 3.9.18 /opt/conda/bin/python

    @pytest.mark.skipif(is_before_spark_320(), reason="only dd/MM/yyyy is supported prior to 3.2.0")
    def test_spark_from_json_date_with_format():
        data = [["""{"time": "26/08/2015"}"""]]
        schema = StructType([StructField("d", DateType())])
>       assert_gpu_and_cpu_are_equal_collect(
                lambda spark : spark.createDataFrame(data, 'json STRING').select(f.col('json'), f.from_json(f.col('json'), schema, {'dateFormat': 'dd/MM/yyyy'})),
            conf = { 'spark.rapids.sql.expression.JsonToStructs': True })

../../src/main/python/json_test.py:1335:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../src/main/python/asserts.py:595: in assert_gpu_and_cpu_are_equal_collect
    _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first, result_canonicalize_func_before_compare=result_canonicalize_func_before_compare)
../../src/main/python/asserts.py:503: in _assert_gpu_and_cpu_are_equal
    from_gpu = run_on_gpu()
../../src/main/python/asserts.py:496: in run_on_gpu
    from_gpu = with_gpu_session(bring_back, conf=conf)
../../src/main/python/spark_session.py:164: in with_gpu_session
    return with_spark_session(func, conf=copy)
/opt/conda/lib/python3.9/contextlib.py:79: in inner
    return func(*args, **kwds)
../../src/main/python/spark_session.py:131: in with_spark_session
    ret = func(_spark)
../../src/main/python/asserts.py:205: in <lambda>
    bring_back = lambda spark: limit_func(spark).collect()
../../../spark-3.3.0-bin-hadoop3/python/pyspark/sql/dataframe.py:817: in collect
    sock_info = self._jdf.collectToPython()
/home/jenkins/agent/workspace/jenkins-rapids_it-non-utc-dev-71/jars/spark-3.3.0-bin-hadoop3/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py:1321: in __call__
    return_value = get_return_value(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

a = ('xro1012184', <py4j.clientserver.JavaClient object at 0x7f0924f23be0>, 'o1012183', 'collectToPython')
kw = {}, converted = IllegalArgumentException()

    def deco(*a: Any, **kw: Any) -> Any:
        try:
            return f(*a, **kw)
        except Py4JJavaError as e:
            converted = convert_exception(e.java_exception)
            if not isinstance(converted, UnknownException):
                # Hide where the exception came from that shows a non-Pythonic
                # JVM exception message.
>               raise converted from None
E               pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.ProjectExec
E               Project [json#111935, from_json(StructField(d,DateType,true), (dateFormat,dd/MM/yyyy), json#111935, Some(Canada/Newfoundland)) AS from_json(json)#111937]
E               +- Scan ExistingRDD[json#111935]

../../../spark-3.3.0-bin-hadoop3/python/pyspark/sql/utils.py:196: IllegalArgumentException
----------------------------- Captured stdout call -----------------------------
### CPU RUN ###
```

--- a/integration_tests/src/main/python/conftest.py
+++ b/integration_tests/src/main/python/conftest.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+# Copyright (c) 2020-2024, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.

--- a/integration_tests/src/main/python/conftest.py
+++ b/integration_tests/src/main/python/conftest.py
@@ -355,6 +355,7 @@ def pytest_collection_modifyitems(config, items):
             if len(qualifier) != 0:
                 qualifier = "_" + qualifier # prefix separator for formatting purposes
         extras.append('DATAGEN_SEED%s=%s' % (qualifier, str(test_datagen_random_seed_choice)))
+        extras.append('TZ=%s' % get_test_tz())
 
         if injection_mode == 'random':
             inject_choice = r.randrange(0, 2) == 1

--- a/integration_tests/src/main/python/json_test.py
+++ b/integration_tests/src/main/python/json_test.py
@@ -743,7 +743,7 @@ non_utc_project_allow = ['ProjectExec'] if is_not_utc() else []
     pytest.param("[0-9]{1,5}", marks=pytest.mark.xfail(reason="https://github.com/NVIDIA/spark-rapids/issues/9588")),
     pytest.param("[1-9]{1,8}", marks=pytest.mark.xfail(reason="https://github.com/NVIDIA/spark-rapids/issues/4940")),
     # floating-point
-    "[0-9]{0,2}\.[0-9]{1,2}"
+    r"[0-9]{0,2}\.[0-9]{1,2}"
     # boolean
     "(true|false)"
 ])
@@ -1328,9 +1328,11 @@ def test_spark_from_json_date_with_locale(data, locale):
         'ProjectExec',
         conf = { 'spark.rapids.sql.expression.JsonToStructs': True })
 
-@pytest.mark.skipif(is_before_spark_320(), reason="only dd/MM/yyyy is supported prior to 3.2.0")
+@allow_non_gpu(*non_utc_allow)
+@pytest.mark.skipif(is_before_spark_320(), reason="dd/MM/yyyy is supported in 3.2.0 and after")
 def test_spark_from_json_date_with_format():
-    data = [["""{"time": "26/08/2015"}"""]]
+    data = [["""{"time": "26/08/2015"}"""],
+            ["""{"time": "01/01/2024"}"""]]
     schema = StructType([StructField("d", DateType())])
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : spark.createDataFrame(data, 'json STRING').select(f.col('json'), f.from_json(f.col('json'), schema, {'dateFormat': 'dd/MM/yyyy'})),

